src/mod.ts

```diff
@@ -96,12 +116,21 @@ export function chatMembers(
         if (!chatId) throw new Error("ctx.chat is undefined and no chatId was provided");
 
         const key = getKey(chatId, userId);
-        const cachedChatMember = await adapter.read(key);
 
-        if (cachedChatMember) return cachedChatMember;
+        const cachedChatMember = enableCaching ? cache.get(key) : undefined;
+        if (cachedChatMember && cachedChatMember.ttl > Date.now()) return cachedChatMember.value;
+        if (cachedChatMember) cache.delete(key);
```

Comment:
Deleting from the cache

Reply:
I was mainly concerned about memory consumption. Since the ttl expired, it makes no sense to keep the record there.
It might be true that it'll get recreated in the next lines, but not if the record has also been deleted from the db and telegram. A non-existing user, for example.
Maybe a bit of over-optimization, but I don't see potential harm.